hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1772,6 +1778,23 @@ void SaraNcpClient::checkRegistrationState() {
 int SaraNcpClient::interveneRegistration() {
     CHECK_TRUE(connState_ == NcpConnectionState::CONNECTING, SYSTEM_ERROR_NONE);
 
+    // Check and store if we are using a Twilio SIM based on ICCID
+    if (!iccidChecked_) {
```

Comment:
(same comment as above)
Nitpicking: Should we move this if condition to a separate function for reusability? This can be a general utility to set iccidChecked_ instead of being only in interveneRegistration()